go/libraries/doltcore/sqle/tables.go

```diff
@@ -1803,3 +1794,69 @@ func (t *AlterableDoltTable) DropPrimaryKey(ctx *sql.Context) error {
 
 	return t.updateFromRoot(ctx, newRoot)
 }
+
+// canDropPk checks whether there are conflicting foreign key relationships referencing this
+// table's primary key tags
+func (t *AlterableDoltTable) verifyCanDropPk(ctx *sql.Context, root *doltdb.RootValue) (*doltdb.ForeignKeyCollection, error) {
+	fkc, err := root.GetForeignKeyCollection(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	indexes := t.sch.Indexes().AllIndexes()
+	if err != nil {
+		return nil, err
+	}
+
+	// pkBackups is a mapping from the table's PL tags to potentially compensating indexes
+	pkBackups := make(map[uint64][]schema.Index, len(t.sch.GetPKCols().TagToIdx))
+	for tag, _ := range t.sch.GetPKCols().TagToIdx {
+		pkBackups[tag] = nil
+	}
+	for _, idx := range indexes {
+		if !idx.IsUserDefined() || !!idx.IsUnique() {
+			continue
+		}
```

Comment:
I'm assuming that we're choosing to not use `UNIQUE` keys as we'd have to check the data, but that's fine. We can short-circuit checks when the key length and fk length are the same as a primary key is a unique key, so we'd only have to check when the primary key is longer than the foreign key.

If unique keys aren't supported then I'd add a comment here stating such. I can clearly see why we only use user-defined indexes, but no idea on why uniques are thrown out.